backend_typescript/server/gameApi.ts

```diff
@@ -0,0 +1,326 @@
+const express = require('express');
+const gameRouter = express.Router();
+import { Failure, Success } from "../models/Result";
+const multer = require('multer');
+const path = require('path');
+import * as db from '../db';
+import * as pf from '../phasesFuncs';
+import { NextFunction, Response } from 'express';
+import User from "../models/User";
+import Game from "../models/Game";
+import { FinalResults } from "../models/GameManager";
+import ShipManager from "../models/ShipManager";
+
+
+interface ProfileUpdateBody {
+    nickname: string;
+    imgId: number | null;
+}
+
+
+interface GameBody {
+    opponent: string;
+    dimension: number;
+}
+
+
+interface ShipBody {
+    ship_size: number;
+    start_row: number;
+    start_col: number;
+    end_row: number;
+    end_col: number;
+}
+
+
+interface ShotBody {
+    row: number;
+    col: number;
+}
+
+
+interface ChatMsg {
+    message: string;
+}
+
+
+// Middlewares
+gameRouter.use(['/', '/profile'], (req, res: Response, next: NextFunction) => {
+    try {
+        if (!req.user) {
+            return res.status(401).json({ msg: 'You need to login first' });
+        }
+        next();
+    } catch (e) {
+        res.status(500).json({ msg: 'Server error' });
+    }
+});
+
+
+
+gameRouter.use('/games/:game_id', async (req, res: Response, next: NextFunction) => {
+    const user = req.user as User;
+    const gameManager = await user.getGameManager();
+    const game = await gameManager.getGameById(req.params.game_id);
+    if (game instanceof Failure) {
+        return res.status(game.status).json({msg: game.msg});
+    }
+    const userGame = game as Success<Game>;
+
+    if (userGame.result.user1 !== user.id && userGame.result.user2 !== user.id) {
```

Comment:
it is better to verify that the user has access to this game inside the function `gameManager.getGameById`